src/patternfly/components/InputGroup/styles.scss

```diff
@@ -29,7 +29,8 @@
     margin-left: -1px;
   }
   /* stylelint-enable */
-  .pf-m-invalid {
+
+  .pf-c-form-control[aria-invalid="true"] {
     margin-left: 0;
 
     &:not(:last-child) {
```

Comment:
same here, I think it looks better without the margin and with `position: relative` (to give it a z-index) so the red border is on top on the right side.